Objects/abstract.c

```diff
@@ -1912,67 +1912,90 @@ PySequence_Tuple(PyObject *v)
         Py_INCREF(v);
         return v;
     }
-    if (PyList_CheckExact(v))
+    
+    if (PyList_CheckExact(v)) {
         return PyList_AsTuple(v);
-
-    /* Get iterator. */
-    it = PyObject_GetIter(v);
-    if (it == NULL)
-        return NULL;
-
+    }
+    
+    // Special cases - End
+    
     /* Guess result size and allocate space. */
-    n = PyObject_LengthHint(v, 10);
-    if (n == -1)
-        goto Fail;
-    result = PyTuple_New(n);
-    if (result == NULL)
-        goto Fail;
-
-    /* Fill the tuple. */
-    for (j = 0; ; ++j) {
-        PyObject *item = PyIter_Next(it);
-        if (item == NULL) {
-            if (PyErr_Occurred())
-                goto Fail;
-            break;
+    Py_ssize_t n = PyObject_LengthHint(v, default_size);
+    
+    PyObject *result = PyTuple_New(n);
+    
+    if (result != NULL) {
+        /* Fill the tuple. */
+        
+        /* Get iterator. */
+        PyObject *it = PyObject_GetIter(v);
+        
+        if (it == NULL) {
+            return NULL;
```

Comment:
here you need to decref `result` otherwise the tuple reference will leak

Reply:
Bella (Thank you)